api/v1alpha1/shared_types.go

```diff
@@ -52,6 +53,11 @@ const (
 
 // KubernetesDeploymentSpec defines the desired state of the Kubernetes deployment resource.
 type KubernetesDeploymentSpec struct {
+	// Merge defines how to perform the merge operation to deployment
+	//
+	// +optional
+	Merge *KubernetesMergeSpec `json:"merge,omitempty"`
```

Comment:
thoughts on calling this field `patch`, so it would look like
```
patch:
  type: StrategicMerge
  spec: 
```

Reply:
Done